pkg/services/ngalert/models/alert_rule.go

```diff
@@ -104,6 +110,27 @@ type AlertRule struct {
 	Labels      map[string]string
 }
 
+// Diff calculates diff between two alert rules. Returns nil if two rules are equal. Otherwise, returns cmputil.DiffReport
+func (alertRule *AlertRule) Diff(rule *AlertRule, ignore ...string) *cmputil.DiffReport {
```

Comment:
Nit: Should we just return a `DiffReport` with no diffs? Why allow this to be `nil`lable too?

Reply:
Thanks for pointing it out. I've come up with a better idea and created DiffReport which is an alias for []Diff so we do not need to return the wrapper.